### ISSUE TITLE
[$250] Workspace - "[Onyx] Provided collection" console errors appear after creating a WS

### ISSUE THREAD

**Issue by lanitochka17** [Expensify/App#42299]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:** 1.4.74.0
**Reproducible in staging?:** Y
**Reproducible in production?:** Y
**If this was caught during regression testing, add the test name, ID and link from TestRail:** N/A
**Issue reported by:** Applause - Internal Team

**Issue found when executing PR** https://github.com/Expensify/App/pull/41651
## Action Performed:
1. Log in with a new expensifail account in Chrome incognito mode
2. Open Chrome console
3. Create a workspace
4. Navigate to any of the default workspace rooms

## Expected Result:
No console errors should appear

## Actual Result:
"Expensify.tsx:43 [Onyx] Provided collection doesn't have all its data belonging to the same parent." console errors appear after creating a WS. Sometimes the errors appear after a few seconds, sometimes it takes 1-2 minutes

## Workaround:
Unknown

## Platforms:

Which of our officially supported platforms is this issue occurring on?
- [ ] Android: Native
- [ ] Android: mWeb Chrome
- [ ] iOS: Native
- [ ] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [ ] MacOS: Desktop

## Screenshots/Videos

Add any screenshot/video evidence

</details> 

https://github.com/Expensify/App/assets/78819774/290894ea-c298-4824-9ed4-7a69b9de306d

[logs (2).txt](https://github.com/Expensify/App/files/15336601/logs.2.txt)

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~014ddfbda5cc9dce3a</li>
        <li>Upwork Job ID: 1792511092144250880</li>
        <li>Last Price Increase: 2024-05-20</li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @shubham1206agra</details>

**Comment by lanitochka17** [comment]
@MitchExpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by MitchExpensify** [comment]
Same error solved here https://expensify.slack.com/archives/C049HHMV9SM/p1689120594593919

**Comment by mountiny** [comment]
Can you please help identify what response from the OpenReport is causing this error @shubham1206agra?

Seems like some of the onyx data might be malformed

**Comment by mountiny** [comment]
I think this is related to this PR https://github.com/Expensify/Auth/pull/10090 that was deployed for a bit.

it causes reports to be duplicated in the mergeCollection call but one set of reports is incorrectly added with 0,1,2,.. indices.

That as a consequence and the mergeCollection as a whole is not applied cc @jasperhuangg 

Thats because [here](https://github.com/Expensify/react-native-onyx/blob/b6a108d98f010462fae42aad2dbbf9baa509959d/lib/Onyx.ts#L409-L410), we are going to mark the entire update as malformed and none of the reports from the mergeCollection will get merged into onyx.

**Comment by jasperhuangg** [comment]
Thank you @mountiny for looking into it. I didn't realize that mergeCollection worked like this and should have been more careful when adding the changes.

The incorrect indices were not intended to be on production at all–a Web PR that was held on the Auth PR was meant to go out shortly afterwards, but was held up on the merge freeze.

**Comment by mountiny** [comment]
This is hard to catch, no problem
